submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/ExtSubmissionSubmitter.kt

```diff
@@ -1,36 +1,60 @@
+    private val queryService: SubmissionPersistenceQueryService,
     private val persistenceService: SubmissionPersistenceService,
     private val draftService: SubmissionDraftService,
     private val requestLoader: SubmissionRequestLoader,
     private val requestProcessor: SubmissionRequestProcessor,
     private val requestReleaser: SubmissionReleaser,
 ) {
-    fun submitAsync(request: SubmissionRequest): Pair<String, Int> {
-        return saveRequest(request, request.submission.submitter)
+    fun createRequest(rqt: ExtSubmitRequest): Pair<String, Int> = createRequest(rqt, rqt.submission.submitter)
+
+    fun loadRequest(accNo: String, version: Int): ExtSubmission = requestLoader.loadRequest(accNo, version)
+
+    fun processRequest(accNo: String, version: Int): ExtSubmission = requestProcessor.processRequest(accNo, version)
+
+    fun checkReleased(accNo: String, version: Int): ExtSubmission = requestReleaser.checkReleased(accNo, version)
+
+    fun release(accNo: String) = requestReleaser.releaseSubmission(accNo)
+
+    fun handleRequest(accNo: String, version: Int): ExtSubmission {
+        return when (queryService.getRequestStatus(accNo, version)) {
+            REQUESTED -> completeRequest(accNo, version)
+            LOADED -> processRequestFiles(accNo, version)
+            FILES_COPIED -> requestReleaser.checkReleased(accNo, version)
+            else -> throw IllegalStateException("Request accNo=$accNo, version='$version' has been already processed")
+        }
     }
 
-    fun processRequest(accNo: String, version: Int): ExtSubmission {
-        val sub = requestLoader.loadRequest(accNo, version)
-        val submission = requestProcessor.processRequest(sub)
-        if (submission.released) requestReleaser.releaseSubmission(accNo)
-        return submission
+    private fun completeRequest(accNo: String, version: Int): ExtSubmission {
+        requestLoader.loadRequest(accNo, version)
+        requestProcessor.processRequest(accNo, version)
+        return requestReleaser.checkReleased(accNo, version)
     }
 
-    fun release(accNo: String) = requestReleaser.releaseSubmission(accNo)
+    private fun processRequestFiles(accNo: String, version: Int): ExtSubmission {
+        requestProcessor.processRequest(accNo, version)
+        return requestReleaser.checkReleased(accNo, version)
+    }
 
-    private fun saveRequest(request: SubmissionRequest, owner: String): Pair<String, Int> {
-        val saved = persistenceService.saveSubmissionRequest(request)
-        request.draftKey?.let { draftService.setProcessingStatus(owner, it) }
+    private fun createRequest(rqt: ExtSubmitRequest, owner: String): Pair<String, Int> {
+        val submission = rqt.submission.copy(version = persistenceService.getNextVersion(rqt.submission.accNo))
+        val saved = persistenceService.createSubmissionRequest(SubmissionRequest(submission, rqt.draftKey, REQUESTED))
+        rqt.draftKey?.let { draftService.setProcessingStatus(owner, it) }
```

Comment:
I just noticed that we are currently just setting the `PROCESSING` status in the draft but we're never setting it to `PROCESSED`or deleting it once the submission is finished. Is this expected behavior? did we miss something? do we care? 😅

Reply:
draft is delete here. https://github.com/EBIBioStudies/biostudies-backend-services/blob/89ac423e2a22c05d7d4cc44fe2d15e6de94c81f9/submission/persistence-mongo/src/main/kotlin/ac/uk/ebi/biostd/persistence/doc/service/ExtSubmissionRepository.kt#L29